src/rviz/image/mouse_click.cpp

```diff
@@ -0,0 +1,127 @@
+#include "rviz/image/mouse_click.h"
+
+#include <ros/names.h>
+
+
+namespace rviz
+{
+
+MouseClick::MouseClick() : QObject()
+{
+}
+
+MouseClick::~MouseClick()
+{
+}
+
+
+void MouseClick::onInitialize(QObject* parent)
+{
+  setParent(parent);
+  parent->installEventFilter(this);
+
+  has_dimensions_ = false;
+  is_topic_name_ok_ = false;
```

Comment:
I would expect these initializations in the constructor already.

Reply:
Ok, makes sense.

Reply:
Done.